lib/improver/utilities/ancillary_creation.py

```diff
@@ -28,12 +28,192 @@
+                alphas_cube.remove_coord(coord)
+        return alphas_cube
+
+    def gradient_to_alpha(self, gradient_x, gradient_y):
+        """
+        Generate alpha smoothing parameters from orography gradients in the
+        x- and y- directions
+
+        Args:
+            gradient_x : iris.cube.Cube
+                A cube of the normalised gradient in the x direction
+            gradient_y : iris.cube.Cube
+                A cube of the normalised gradient in the y direction
+
+        Returns:
+            alpha_x : iris.cube.Cube
+                A cube of alphas in the x direction
+            alpha_y : iris.cube.Cube
+                A cube of alphas in the y direction
+        """
+        alpha_x = self.unnormalised_alphas(gradient_x)
+        alpha_y = self.unnormalised_alphas(gradient_y)
+
+        if self.invert_alphas:
+            alpha_x, alpha_y = self.scale_alphas([alpha_x, alpha_y],
+                                                 min_output=self.max_alpha,
+                                                 max_output=self.min_alpha)
+        else:
+            alpha_x, alpha_y = self.scale_alphas([alpha_x, alpha_y],
+                                                 min_output=self.min_alpha,
+                                                 max_output=self.max_alpha)
+        alpha_x = self.update_alphas_metadata(alpha_x)
+        alpha_y = self.update_alphas_metadata(alpha_y)
+
+        return alpha_x, alpha_y
+
+    def process(self, cube):
+        """
+        This creates the alpha cubes. It returns one for the x direction and
+        one for the y direction. It uses the
+        DifferencBetweenAdjacentGridSquares plugin to get the height
+        difference between grid spaces and then calculates a gradient,
+        which is normalised between between numbers (which can be chosen).
+        The gradients are then linearly regridded so that they match the
+        orography dimensions and will go into the recursive filter.
+
+        Args:
+            cube : iris.cube.Cube
+                A 2D cube of the orography for the grid we want to get alphas
+                for.
+
+        Returns:
+            alpha_x : iris.cube.Cube
+                A cube of orographic dependent alphas calculated in the x
+                direction.
+            alpha_y : iris.cube.Cube
+                A cube of orographic dependent alphas calculated in the y
+                direction.
+        """
+        if not isinstance(cube, iris.cube.Cube):
+            raise ValueError('OrographicAlphas() expects cube input, got {}'
+                             .format(type(cube)))
+
+        if len(cube.data.shape) != 2:
+            raise ValueError('Expected orography on 2D grid, got {} dims'
+                             .format(len(cube.data.shape)))
+
+        gradient_x, gradient_y = \
+            DifferenceBetweenAdjacentGridSquares(gradient=True).process(cube)
```

Comment:
Insert this before your call to gradient_to_alpha to enable you to remove abs() in the more general gradient_from_diff function below. See comments there for why this helps you in another way.
```
        gradient_x.data = abs(gradient_x.data)
        gradient_y.data = abs(gradient_y.data)
````

Reply:
Have updated the functionality of gradients_to_alphas to include the "abs".